src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxrResolver.cpp

```diff
@@ -121,7 +158,12 @@ HostFxrResolver::GetHostFxrParameters(
                 {
                     dotnetExePath = GetAbsolutePathToDotnet(applicationPhysicalPath, L"dotnet");
                 }
-                hostFxrDllPath = GetAbsolutePathToHostFxr(dotnetExePath);
+
+                fs::path dotnetExePathParent = dotnetExePath.parent_path();
+                if (!TryGetHostFxrPath(hostFxrDllPath, dotnetExePathParent, applicationPhysicalPath))
+                {
+                    throw InvalidOperationException(format(L"get_hostfxr_path failed"));
```

Comment:
Does this exception get surfaced to the user? Might be worth plumbing through the return value from get_hostfxr_path?

Reply:
Also, just a thought: do we want to tell people how to collect the traces from nethost when stuff goes wrong?

Reply:
That's actually the last set of tests that are broken right now, CoreHostTrace is the only way you can see what's going wrong since, we already log the return value in the logs, but I can return an HR instead of a bool so we have the code